BDD Scenario Create

Kristin01 opened the pull request:
Added a scenario for Create endpoint, and added active field into db and modified related files

liondon commented on features/steps/promotion_steps.py:
```diff
@@ -138,10 +125,6 @@ def step_impl(context, element_name):
-
-##################################################################
-# These two function simulate copy and paste
-##################################################################
```
I'm thinking maybe we could leave these comments?

liondon commented on features/steps/promotion_steps.py:
```diff
@@ -138,10 +125,6 @@ def step_impl(context, element_name):
-
-##################################################################
-# These two function simulate copy and paste
-##################################################################
```
I'm thinking maybe we could leave these comments? So that whoever needs to work on these could understand the code easier?

Kristin01 replied:
@liondon Hey Ling-I, I add the `active` filed in DB model cause according to create function in`rest_api.js`, we can set the `active` checkbox, that means it would generate an `active` data when creating a promotion, which needs a place to store in the database. 
And about change files in tests, since we add a new field(`active`) in DB model, there are some functions, like serialize, deserialize functions and so on, related to the `active` filed, so I think we need change these relevant files to make the tests work.

Kristin01 replied:
@liondon I think if we do not want to include  the `active` filed in DB model, we could cancel the feature of setting `active` checkbox in create function( and could also cancel this in update function I think), and only leave the feature of setting `active` checkbox in search function.

liondon replied:
@Kristin01 yeah, I think the checkbox is just there for search, maybe we could just ignore the checkbox when we create promo in `rest_pi.js` script? @VickySteeves What do you think?

Kristin01 replied:
@liondon Yeah, and I think we could also ignore`update_form_data` in `rest_pi.js`.

Kristin01 replied:
@liondon @VickySteeves If we agree on the above changes ( to cancel  the feature of setting active checkbox (right now it changes to dropdown) of create, update promotion and `update_form_data` in `rest_pi.js`), I will make a modification now. Is that Okay?

liondon commented on features/steps/promotion_steps.py:
```diff
@@ -162,14 +145,6 @@ def step_impl(context, element_name):
-# in lowercase followed by '-btn' so the Clean button has an id of
-# id='clear-btn'. That allows us to lowercase the name and add '-btn'
-# to get the element id of any button
-##################################################################
```
the same.

Kristin01 replied:
Agreed.

liondon commented on features/steps/promotion_steps.py:
```diff
@@ -182,14 +157,6 @@ def step_impl(context, message):
-# The id field for text input in the html is the element name
-# prefixed by ID_PREFIX so the Name field has an id='pet_name'
-# We can then lowercase the name and prefix with pet_ to get the id
-##################################################################
```
the same.

liondon reviewed (changes requested):
Hey Kristin, I think we're not including the `active` filed in our model because it's redundant. The way we check if a promo is active or not is by checking its `start_date` and `end_date`. Is there a particular reason you need to deal with this for the story?

Also, I'm not sure why do we need to change files in `tests`? I think our BDD tests and TDD tests are totally irrelevant?

Kristin01 commented:
Updated the info:
This PR now has the following changes:
1. Added a scenario for Create endpoint.
2. Added some sample data to create more query scenarios
3. Cancel the feature of selecting active dropdown of create and update functions in `rest_api.js` to make the create (and update) scenario tests pass.
4. Fixed code to make the tests work.

liondon reviewed (approved):
Looks great!

liondon commented:
IBM Cloud [toolchain](https://cloud.ibm.com/devops/toolchains/82ed68f3-a995-419e-80a4-f9191b98d0a4?env_id=ibm:yp:us-south "nyu-devops-promo-toolchain"): [Delivery Pipeline](https://cloud.ibm.com/devops/pipelines/a90cf2f8-b961-4e4c-98ff-49891084d12b "nyu-devops-promo-pipeline") deployed [promotions](https://github.com/2020-nyu-devops/promotions/releases/tag/deploy-ibm-yp-us-south-2020-nyu-devops-dev-20201122-075601 "deploy-ibm-yp-us-south-2020-nyu-devops-dev-20201122-075601") to [dev](https://console.bluemix.net/apps/42c5c4c9-a7f2-472a-bfb1-add0f77b8a15?env_id=ibm:yp:us-south "ibm:yp:us-south:2020-nyu-devops:dev"), including commit 48a816b00e8091f88a49ae0773ca56e1b0b2b81d

liondon commented:
IBM Cloud [toolchain](https://cloud.ibm.com/devops/toolchains/82ed68f3-a995-419e-80a4-f9191b98d0a4?env_id=ibm:yp:us-south "nyu-devops-promo-toolchain"): [Delivery Pipeline](https://cloud.ibm.com/devops/pipelines/a90cf2f8-b961-4e4c-98ff-49891084d12b "nyu-devops-promo-pipeline") deployed [promotions](https://github.com/2020-nyu-devops/promotions/releases/tag/deploy-ibm-yp-us-south-2020-nyu-devops-prod-20201122-080252 "deploy-ibm-yp-us-south-2020-nyu-devops-prod-20201122-080252") to [prod](https://console.bluemix.net/apps/30a9e46b-e89b-4c2e-930e-53e6242584de?env_id=ibm:yp:us-south "ibm:yp:us-south:2020-nyu-devops:prod"), including commit 48a816b00e8091f88a49ae0773ca56e1b0b2b81d
